### PR TITLE
feat(call-home): make product name configurable

### PR DESCRIPTION
**Pull request by niladrih**

### CODE REVIEW

File: call-home/src/common/constants.rs [hunk 1]
```diff
@@ -6,7 +6,10 @@ use utils::version_info;
 
 /// PRODUCT is the name of the project for which this call-home component is deployed.
 pub fn product() -> String {
-    ::constants::product_train()
+    env::var("PRODUCT_NAME")
+        .ok()
+        .filter(|v| !v.is_empty())
+        .unwrap_or(::constants::product_train())
```
Comment (dsharma-dc) [inline call-home/src/common/constants.rs]: Is this working well for both - when env var is not set and also when it is set to blank "" ?

**Review by dsharma-dc — APPROVED** [review]

File: call-home/src/common/constants.rs (hunk 1 above)
Comment (niladrih) [reply call-home/src/common/constants.rs]: Yeah, it is. `env::var()` returns a `Result::Err()` for unset env, but an `Ok()` for empty env. That's why there's a `filter(|v| !v.is_empty())` to cut that case out as a `None()`.

I'll add a unit test, as I had the same concern.

Comment (niladrih) [reply call-home/src/common/constants.rs]: Added a unit test.

File: call-home/src/common/constants.rs [hunk 2]
```diff
@@ -6,7 +6,10 @@ use utils::version_info;
 
 /// PRODUCT is the name of the project for which this call-home component is deployed.
 pub fn product() -> String {
-    ::constants::product_train()
+    env::var("PRODUCT_NAME")
```
Comment (tiagolobocastro) [inline call-home/src/common/constants.rs]: What is the reason for this change?

File: call-home/src/common/constants.rs [hunk 3]
```diff
@@ -4,9 +4,15 @@ use std::{
...
 
+/// CALLHOME_PRODUCT_NAME_ENV is the name of the ENV which configures the product-name for callhome.
+pub const CALLHOME_PRODUCT_NAME_ENV: &str = "CALLHOME_PRODUCT_NAME";
+
 /// PRODUCT is the name of the project for which this call-home component is deployed.
 pub fn product() -> String {
-    ::constants::product_train()
+    env::var(CALLHOME_PRODUCT_NAME_ENV)
+        .ok()
+        .filter(|v| !v.is_empty())
```
Comment (tiagolobocastro) [inline call-home/src/common/constants.rs:15]: Need to map to train as well

Comment (niladrih) [reply call-home/src/common/constants.rs:15]: Okay..

**Review by tiagolobocastro — APPROVED** [review]

**Comment by niladrih** [comment]
bors merge
